packages/iov-bns/src/decode.ts

```diff
@@ -104,37 +100,11 @@ export function decodeAmount(coin: codecImpl.coin.ICoin): Amount {
   };
 }
 
-// adds zeros to the right as needed to ensure given length
-function rightPadZeros(short: string, length: number): string {
-  if (short.length >= length) {
-    return short;
-  }
-  return short + "0".repeat(length - short.length);
-}
-
-// we only allow up to 9 decimal places
-const humanCoinFormat = new RegExp(/^(\d+)(\.\d{1,9})?\s*([A-Z]{3,4})$/);
-
-export function decodeJsonAmount(json: string): Amount {
-  const data = JSON.parse(json);
-  if (typeof data === "string") {
-    // parse eg. "1.23 IOV"
-    const vals = humanCoinFormat.exec(data);
-    if (vals === null) {
-      throw new Error(`Invalid coin string: ${data}`);
-    }
-    const [, wholeStr, fracString, ticker] = vals;
-    const coin = {
-      whole: Int53.fromString(wholeStr).toNumber(),
-      fractional: fracString ? Int53.fromString(rightPadZeros(fracString.slice(1), 9)).toNumber() : undefined,
-      ticker: ticker as TokenTicker,
-    };
-    return decodeAmount(coin);
-  } else if (typeof data === "object" && data !== null) {
-    // parse a json coin representation
-    return decodeAmount(data as codecImpl.coin.ICoin);
-  }
-  throw new Error("Impossible type for amount json");
+export function decodeCashConfiguration(config: codecImpl.cash.Configuration): CashConfiguration {
```

Comment:
Glad to see decodeJsonAmount gone... this is simpler, no?

Reply:
yes simpler, and most of all: consistent. Everything I get from the chain is provable (thus binary) encoded as protobuf.